Remotion/Validation/Globalization.UnitTests/LocalizedValidationMessageFactoryTest.cs

```diff
@@ -217,43 +216,62 @@ public void CreateValidationMessageForPropertyValidator_WithMinimumLengthValidat
     {
-      _propertyStub.Setup(_ => _.PropertyType).Returns(typeof(Assert));
-      var validator = new NotEmptyValidator(_validationMessageStub.Object);
+      _propertyStub.Setup(_ => _.PropertyType).Returns(typeof(byte[]));
+      var validator = new NotEmptyBinaryValidator(_validationMessageStub.Object);
       var validationMessage = _factory.CreateValidationMessageForPropertyValidator(validator, _propertyStub.Object);
 
-      Assert.That(validationMessage, Is.Null);
+      Assert.That(validationMessage, Is.Not.Null);
+      Assert.That(validationMessage, Is.InstanceOf<ResourceManagerBasedValidationMessage>());
+
+      Assert.That(validationMessage.ToString(), Is.EqualTo("Enter a value."));
     }
 
     [Test]
-    public void CreateValidationMessageForPropertyValidator_WithNotEmptyValidatorAndPropertyIsString_ReturnsLocalizedValidationMessage ()
+    public void CreateValidationMessageForPropertyValidator_WithNotEmptyCollectionValidator_WithReadOnlyCollection_ReturnsLocalizedValidationMessage ()
     {
-      _propertyStub.Setup(_ => _.PropertyType).Returns(typeof(string));
-      var validator = new NotEmptyValidator(_validationMessageStub.Object);
+      _propertyStub.Setup(_ => _.PropertyType).Returns(typeof(IReadOnlyCollection<object>));
+      var validator = new NotEmptyCollectionValidator(_validationMessageStub.Object);
       var validationMessage = _factory.CreateValidationMessageForPropertyValidator(validator, _propertyStub.Object);
 
       Assert.That(validationMessage, Is.Not.Null);
       Assert.That(validationMessage, Is.InstanceOf<ResourceManagerBasedValidationMessage>());
 
-      Assert.That(validationMessage.ToString(), Is.EqualTo("Enter a value."));
+      Assert.That(validationMessage.ToString(), Is.EqualTo("Add an entry."));
     }
 
     [Test]
-    public void CreateValidationMessageForPropertyValidator_WithNotEmptyValidatorAndPropertyIsIEnumerable_ReturnsLocalizedValidationMessage ()
+    public void CreateValidationMessageForPropertyValidator_WithNotEmptyCollectionValidator_WithCollection_ReturnsLocalizedValidationMessage ()
     {
-      _propertyStub.Setup(_ => _.PropertyType).Returns(typeof(IEnumerable));
-      var validator = new NotEmptyValidator(_validationMessageStub.Object);
+      _propertyStub.Setup(_ => _.PropertyType).Returns(typeof(ICollection));
+      var validator = new NotEmptyCollectionValidator(_validationMessageStub.Object);
+      var validationMessage = _factory.CreateValidationMessageForPropertyValidator(validator, _propertyStub.Object);
+
+      Assert.That(validationMessage, Is.Not.Null);
+      Assert.That(validationMessage, Is.InstanceOf<ResourceManagerBasedValidationMessage>());
+
+      Assert.That(validationMessage.ToString(), Is.EqualTo("Add an entry."));
+    }
+
+    [Test]
+    public void CreateValidationMessageForPropertyValidator_WithNotEmptyCollectionValidator_WithGenericCollection_ReturnsLocalizedValidationMessage ()
```

Comment:
I hope I'm not contradicting my first review here: for the MessageFactory, we don't need to cover individual property types supported by the validator, it should be enough to have one test for the validator-type to achieve coverage (the factory does not know about the property type).

Reply:
It does contradict the earlier review.
`LocalizedValidationMessageFactory` get the `IPropertyInformation` as input parameter, so it can differentiate by property type. For `NotEmptyCollectionValidation`, there is only one path in the implementation, but it could be different (like for `LengthValidator` and `NotNullValidator`), wherefore the additional tests make sense.

Reply:
Ah, okay, yes, thank you. 

okay, so it comes down to if you do htis test first, you see there's no special logic for propertytype needed with collection validatior and you'd stop with one test. It's okay to cover all three, unless we think it could confuse someone reading the tests. (=> bascally, it doesn't matter if there's one or three in this particular instance)